app/controllers/admin/supplier_users_controller.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+class Admin::SupplierUsersController < Admin::BaseController
+  skip_after_action :verify_authorized, only: :index
+  skip_after_action :verify_policy_scoped, except: :index
+
+  before_action :set_suppliers, only: %i[new receive_supplier]
+  before_action :form_from_session, only: %i[user_details review create success]
+
+  def index
+    lead_providers = policy_scope(LeadProvider).joins(:users).includes(:users)
+    delivery_partners = policy_scope(DeliveryPartner).joins(:users).includes(:users)
+    sorted_users = (lead_providers + delivery_partners).flat_map(&:users).sort_by(&:full_name)
+    @users = Kaminari.paginate_array(sorted_users).page(params[:page]).per(20)
+    @page = @users.current_page
+    @total_pages = @users.total_pages
+  end
+
+  def new
+    authorize User, :new?
```

Comment:
It's not just user that we want to authorise here right? We are gonna create some profiles as well, and that's really the part that needs authorisation.

Reply:
You happy for that to go in the create? When we know what type of profile we are creating

Reply:
Did I mention how much I hate that we have different entities being managed in the same place?

Fine, create is a good place for that.